astropy/cosmology/flrw.py

```diff
@@ -438,20 +436,17 @@ def Ob(self, z):
         ValueError
           If Ob0 is None.
         """
-
         if self._Ob0 is None:
             raise ValueError("Baryon density not set for this cosmology")
-        if isiterable(z):
-            z = u.Quantity(z, u.dimensionless_unscaled).value
-        return self._Ob0 * (1. + z) ** 3 * self.inv_efunc(z) ** 2
+        return self._Ob0 * (z + 1.0) ** 3 * self.inv_efunc(z) ** 2
 
     def Odm(self, z):
         """ Return the density parameter for dark matter at redshift ``z``.
 
         Parameters
         ----------
-        z : array-like
-          Input redshifts.
+        z : `~astropy.units.Quantity` ['redshift'], `~numbers.Number`, or ndarray
```

Comment:
Could this become `Quantity-like`?

Reply:
I have PR #12177 to clean all this up. I'll rebase this PR when that is merged.